src/fhirtypes/ElementDefinition.ts

```diff
@@ -238,6 +240,290 @@ export class ElementDefinition {
     [this.min, this.max] = [min, max];
   }
 
+  /**
+   * Constrains the type of this element to the requested type(s).  When this element's type is a
+   * choice, this function will reduce the choice to only those types provided -- unless a
+   * targetType is provided, in which case, only that type will be affected and other options in
+   * the choice will be left unchanged.  This function should allow the following scenarios:
+   * - constrain a choice of types to a smaller subset of types (including a single type)
+   * - constrain a type to one or more profiles on that type
+   * - constrain a supertype (e.g., Resource) to one or more subtypes (e.g., Condition)
+   * - constrain a reference of multiple types to a reference of a smaller subset of types
+   * - constrain a reference of a type or profile to one or more profiles of that type/profile
+   * - constrain a reference of a supertype to one or more references of subtypes
+   * - any combinaton of the above
+   * This function will throw when:
+   * - attempting to add a base type (e.g., `type.code`) that wasn't already a choice in the type
+   * - attempting to add a profile that doesn't match any of the existing types
+   * - attempting to add a base reference that wasn't already a reference
+   * - attempting to add a reference to a profile that doesn't match any of the existing references
+   * - specifying a target that does not match any of the existing type choices
+   * - specifying types or a target whose definition cannot be found
+   * @see {@link http://hl7.org/fhir/R4/elementdefinition-definitions.html#ElementDefinition.type}
+   * @param {{ type: string; isReference?: boolean }[]} types - the set of constrained types,
+   *   identified by id/type/url strings and an optional reference flag (defaults false)
+   * @param {ResolveFn} resolve - a function that can resolve a type to a StructureDefinition instance
+   * @param {string} [target] - a specific target type to constrain.  If supplied, will attempt to
```

Comment:
Just a nitpick, but is there a reason that `target` is in brackets here? Is that some notation I don't understand, or is it a mistake?

Reply:
In JSDoc, you can use brackets to indicate that an argument is optional or can be null.